dll/win32/shell32/CShellLink.cpp

```diff
@@ -2619,11 +2619,14 @@ HRESULT STDMETHODCALLTYPE CShellLink::QueryContextMenu(HMENU hMenu, UINT indexMe
 
 HRESULT CShellLink::DoOpenFileLocation()
 {
+    WCHAR szExplorer[MAX_PATH];
+    Shell_GetShellProgram(szExplorer, _countof(szExplorer));
+
     WCHAR szParams[MAX_PATH + 64];
     StringCbPrintfW(szParams, sizeof(szParams), L"/select,%s", m_sPath);
 
     INT_PTR ret;
-    ret = reinterpret_cast<INT_PTR>(ShellExecuteW(NULL, NULL, L"explorer.exe", szParams,
+    ret = reinterpret_cast<INT_PTR>(ShellExecuteW(NULL, NULL, szExplorer, szParams,
```

Comment:
I don't think this is right in all cases. For example, this instance refers to explorer the file browser, not the shell. Trying to open a file browser by calling an alternative shell is unlikely to work.

The correct way to do this is is likely to use the appropriate DDE & COM interfaces to interact with the shell process and request a folder view. So while that stuff isn't fully implemented, leaving "explorer" for now is probably the best thing we can do.